### PR TITLE
T18383

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T18383

### CODE REVIEW

File: js/ui/overview.js [hunk 1]
```diff
@@ -698,6 +701,9 @@ const Overview = new Lang.Class({
         if (!this._syncGrab())
             return;
 
+        log("Showing overview");
+        log(new Error().stack);
```
Comment (cosimoc) [inline js/ui/overview.js]: Remove

File: js/ui/overview.js [hunk 2]
```diff
@@ -774,6 +780,9 @@ const Overview = new Lang.Class({
         if (!this._shown)
             return;
 
+        log("Hiding overview");
+        log(new Error().stack);
```
Comment (cosimoc) [inline js/ui/overview.js]: Remove

File: js/ui/overview.js [hunk 3]
```diff
@@ -533,6 +533,9 @@ const Overview = new Lang.Class({
         if (this.isDummy)
             return;
 
+        if (Main.wm.windowsAreMappedAndVisible())
```
Comment (cosimoc) [inline js/ui/overview.js]: Can you not use `Main.workspaceMonitor.hasActiveWindows()` which is used elsewhere in `overview.js` for similar purposes?

File: js/ui/overview.js (hunk 1 above)
Comment (smspillaz) [inline js/ui/overview.js]: Yikes, I thought I got rid of these

File: js/ui/overview.js (hunk 3 above)
Comment (smspillaz) [inline js/ui/overview.js]: Yep, that seems to work - thanks for the pointer!

**Comment by cosimoc** [comment]
This looks good, but please squash the commits together before we merge it.

**Pull request merged**